core/keychain/keychain.go

```diff
@@ -137,7 +203,7 @@ func (kc *keychain) generateKeyPair(seed []byte) ([]byte, []byte, error) {
 	return pub, priv, err
 }
 
-func (kc *keychain) generateAndStoreKeyPair(seed []byte) ([]byte, []byte, error) {
+func (kc *keychain) generateAndStoreKeyPair(seed []byte, mnemonic string) ([]byte, []byte, error) {
```

Comment:
Can someone pass in an uncorrelated `seed` and `mnemonic` here? Maybe it just gets one and the other is generated so there is less room for error? Edit: it could be updated later too since we only use it in one place rn.

Reply:
This is an auxiliary/private method called after generating the mnemonic / seed so feel it's fair to assume it's not going to be misused. Can't generate one from the other since it's being used when generating keys without mnemonic. Agree we need to update it later since generating keys without mnemonic might be deprecated and we can clean up the keychain class.